pkg/handler/collector/oci/oci_registry.go

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oci
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/guacsec/guac/pkg/collectsub/datasource"
+	"github.com/guacsec/guac/pkg/collectsub/datasource/inmemsource"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/pkg/errors"
+	"github.com/regclient/regclient"
+	"github.com/regclient/regclient/types"
+	"github.com/regclient/regclient/types/ref"
+)
+
+const (
+	OCIRegistryCollector = "OCIRegistryCollector"
+)
+
+type ociRegistryCollector struct {
+	collectDataSource datasource.CollectSource
+	checkedDigest     map[string][]string
+	registry          string
+	poll              bool
+	interval          time.Duration
+}
+
+func NewOCIRegistryCollector(ctx context.Context, registry string, poll bool, interval time.Duration) *ociRegistryCollector {
+	return &ociRegistryCollector{
+		checkedDigest: map[string][]string{},
+		registry:      registry,
+		poll:          poll,
+		interval:      interval,
+	}
+}
+
+func (o *ociRegistryCollector) RetrieveArtifacts(ctx context.Context, docChannel chan<- *processor.Document) error {
+	r, err := ref.New(o.registry)
+	if err != nil {
+		return fmt.Errorf("failed to parse ref %s: %v", r, err)
+	}
+
+	rcOpts := getRegClientOptions()
+	rc := regclient.New(rcOpts...)
+	defer rc.Close(ctx, r)
```

Comment:
Close can return an error. This would not be noticed if someone is trying to debug.

A recommendation would be to log this.

```golang
defer func(rc *regclient.RegClient, ctx context.Context, r ref.Ref) {
		err := rc.Close(ctx, r)
		if err != nil {
			//log error
		}
	}(rc, ctx, r)
```

Reply:
This has been handled in the new PR